src/redux/features/AuthSlice/EditorSlice.ts

```diff
@@ -1,38 +1,72 @@
-import { createAction, createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit'
+import {
+  createAction,
+  createAsyncThunk,
+  createSlice,
+  PayloadAction,
+  SerializedError,
+} from '@reduxjs/toolkit'
 import { RootState } from '../../store'
 import { HYDRATE } from 'next-redux-wrapper'
 import client from '@/apollo'
 import { gql } from '@apollo/client'
 
 const hydrate = createAction<RootState>(HYDRATE)
 
-type TUnknownObject = Record<string, string | number | string | Array<TUnknownObject>>
-
-type EditorState = {
-  query: string
-  data: TUnknownObject
-}
+type TUnknownObject = Record<string, string | number | string | Array<TUnknownObject>> | string
 
 export const CHARACTERS_QUERY = `
-  query getCharacters {
-    characters {
-      results {
-        name
-      }
+query getCharacters($filter: FilterCharacter) {
+  characters(filter: $filter) {
+    results {
+      name
+      status
     }
   }
+}
+`
+
+export const CHARACTERS_VARS = `{
+  "filter": {
+    "status": "Alive"
+  }
+}
+`
+
+export const CHARACTERS_HEADERS = `{
+  "special": "Special header value"
+}
 `
 
+type EditorState = {
+  query: string
+  variables: string
+  headers: string
+  data: TUnknownObject
+  status: 'idle' | 'loading'
+  error: null | SerializedError
+}
+
 const initialState: EditorState = {
   query: CHARACTERS_QUERY,
+  variables: CHARACTERS_VARS,
+  headers: CHARACTERS_HEADERS,
   data: {},
+  status: 'idle',
+  error: null,
 }
 
-export const getGqlValueThunk = createAsyncThunk('editor/getGqlQuery', async (query: string) => {
+export const getGqlValueThunk = createAsyncThunk('editor/getGqlQuery', async (_, thunkAPI) => {
+  const { query, variables, headers } = (thunkAPI.getState() as RootState).editorSlice
+
   const response = await client.query({
     query: gql`
       ${query}
     `,
+
```

Comment:
unnecessary empty line

Reply:
will be fixed/refactor in following commits